projects/ngx-qgrid-plugins/src/lib/pane/pane.component.ts

```diff
@@ -35,37 +39,43 @@ export class PaneComponent implements OnInit {
 			observeReply(model[`${state}Changed`])
 				.subscribe((e: GridEventArg<any>) => {
 					if (!prop || e.hasChanges(prop)) {
-						this.close('right');
-						this.open('right');
+						this.updateAndOpen('right')
 					}
 				});
 		}
 	}
 
-	open(side: 'right') {
+	open(side: PaneSide, value: any = null) {
 		const { table, model } = this.plugin;
 
-		let value = null;
 		const scope = this.parse();
-		if (scope) {
+		if (scope && !value) {
 			const [state, prop] = scope;
 			value = model[state]()[prop];
 		}
 
-		this.context = { $implicit: this, value };
+		this.context[side] = { $implicit: this, value };
 		table.view.addLayer(`pane-${side}`);
 
-		this.cd.markForCheck();
-		this.cd.detectChanges();
+		this.invalidate();
 	}
 
-	close(side: 'right') {
+	close(side: PaneSide) {
 		const { table } = this.plugin;
 
 		table.view.removeLayer(`pane-${side}`);
 
-		this.context = null;
+		this.context[side] = null;
 
+		this.invalidate();
+	}
+
+	updateAndOpen(side: PaneSide, value: any = null) {
```

Comment:
why we need that method? as I understood just call `open` should be enough in the example code?

Reply:
nope. It will not re-render existing pane.
You had the same code previously in ngOnInit() on trigger
![image](https://user-images.githubusercontent.com/33963441/107382352-d1dc9b80-6b00-11eb-8c9d-7d12628a523b.png)